lib/vsc/filesystem/oceanstor.py

```diff
@@ -430,38 +612,52 @@ def _local_filesystems(self):
             self.list_filesystems()
 
         # Add filesystem name in OceanStor to list of attributes of local filesystems
-        self.localfilesystemnaming.append('oceanstorfs')
+        self.localfilesystemnaming.append('oceanstor')
+
+        # NFS share paths and their IDs
+        oceanstor_shares = self.list_nfs_shares()
+        oceanstor_share_paths = {
+            os.path.normpath(nfs_share[ns]['share_path']): nfs_share[ns]['dtree_id']
+            for nfs_share in oceanstor_shares.values()
+            for ns in nfs_share
+        }
 
         for fs in self.localfilesystems:
-            fs_oceanstor_name = None
+            oceanstor_tag = None
 
             if fs[self.localfilesystemnaming.index('type')] in self.supportedfilesystems:
-                # Check NFS mount server and filesystem name
+                # Possible NFS mount from OceanStor
+                mount_point = fs[self.localfilesystemnaming.index('mountpoint')]
                 mount_device = fs[self.localfilesystemnaming.index('device')]
-                mount_ip, fs_path = mount_device.split(':', 1)
-
-                # Determine if filesystem is a known filesystem in OceanStor
-                if mount_ip in self.nfs_ips:
-                    fs_name = os.path.basename(fs_path)
-                    try:
-                        fs_oceanstor_name = self.oceanstor_filesystems[fs_name]['name']
-                    except KeyError as err:
-                        errmsg = "NFS mount '%s' served from OceanStor has an unkown filesystem '%s' to the REST API"
-                        self.log.raiseException(errmsg % (mount_device, fs_name), OceanStorOperationError)
+                server_ip, share_path = mount_device.split(':', 1)
+
+                # Check NFS server IP
+                server_ip = IPv4Address(server_ip)
+                oceanstor_nfs_servers = self.list_nfs_servers()
+                if any(server_ip == oceanstor_ip for oceanstor_ip in oceanstor_nfs_servers):
+                    # Check share path
+                    share_path = os.path.normpath(share_path)
+                    if share_path in oceanstor_share_paths:
+                        oceanstor_tag = oceanstor_share_paths[share_path]
+                        dbgmsg = "Local NFS mount '%s' is served by OceanStor and shares object ID: %s"
+                        self.log.debug(dbgmsg, mount_point, oceanstor_tag)
+                    else:
+                        errmsg = "NFS mount '%s' served from OceanStor '%s' shares unkown path '%s'"
```

Comment:
```suggestion
                        errmsg = "NFS mount '%s' served from OceanStor '%s' shares unknown path '%s'"
```

Reply:
fixed in https://github.com/vub-hpc/vsc-filesystem-oceanstor/pull/4/commits/690c952d1cc3f210e1060790adf82aec0489f90e